wandb/sdk/internal/internal_api.py

```diff
@@ -1365,35 +1390,64 @@ def get_project_run_queues(self, entity: str, project: str) -> List[Dict[str, st
 
     @normalize_exceptions
     def create_default_resource_config(
-        self, entity: str, resource: str, config: str
+        self,
+        entity: str,
+        resource: str,
+        config: str,
+        template_variables: Optional[Dict[str, Union[float, int, str]]],
     ) -> Optional[Dict[str, Any]]:
         if not self.create_default_resource_config_introspection():
             raise Exception()
-        query = gql(
-            """
-        mutation createDefaultResourceConfig(
-            $entityName: String!
-            $resource: String!
+        supports_template_vars = self.push_to_run_queue_introspection()
+
+        mutation_params = """
+            $entityName: String!,
+            $resource: String!,
             $config: JSONString!
-        ) {
-            createDefaultResourceConfig(
-            input: {
-                entityName: $entityName
-                resource: $resource
-                config: $config
-            }
-            ) {
-            defaultResourceConfigID
-            success
-            }
-        }
         """
-        )
+        mutation_inputs = """
+            entityName: $entityName,
+            resource: $resource,
+            config: $config
+        """
+
+        if supports_template_vars:
+            mutation_params += ", $templateVariables: JSONString"
+            mutation_inputs += ", templateVariables: $templateVariables"
+        else:
+            if template_variables is not None:
+                raise UnsupportedError(
+                    "server does not support template variables, please update server instance"
```

Comment:
do you know minimal version where it does? probably would be more useful if the user knows what to update to. not required, just a thought

Reply:
Nice, yeah I can add that